From e65008c95fe8ef46218b323e1fe27d4f965c7000 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 07 Aug 2018 16:39:24 +0300
Subject: [PATCH] Add BTL support Fixup [PCIe PHY FW]

Remove external FW sectors since it is embedded on BTL images
---

diff --git a/board/habanalabs/dali/btl/Makefile b/board/habanalabs/dali/btl/Makefile
index 11500d6..66399f4 100644
--- a/board/habanalabs/dali/btl/Makefile
+++ b/board/habanalabs/dali/btl/Makefile
@@ -18,16 +18,12 @@
 endif
 
 PRIMARY_CPP_FLAGS += $(COMMON_CPP_FLAGS)
-PRIMARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x1b0000
-PRIMARY_CPP_FLAGS += -DPCIE_PHY_FW_SIZE=$(PCIE_PHY_FW_SIZE)
 PRIMARY_CPP_FLAGS += -DSPL_OFFSET=0xa0000
 PRIMARY_CPP_FLAGS += -DSPL_SIZE=$(SPL_SIZE)
 PRIMARY_CPP_FLAGS += -DUBOOT_OFFSET=0xb0000
 PRIMARY_CPP_FLAGS += -DUBOOT_SIZE=$(UBOOT_SIZE)
 
 SECONDARY_CPP_FLAGS += $(COMMON_CPP_FLAGS)
-SECONDARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x9b0000
-SECONDARY_CPP_FLAGS += -DPCIE_PHY_FW_SIZE=$(PCIE_PHY_FW_SIZE)
 SECONDARY_CPP_FLAGS += -DSPL_OFFSET=0x8a0000
 SECONDARY_CPP_FLAGS += -DSPL_SIZE=$(SPL_SIZE)
 SECONDARY_CPP_FLAGS += -DUBOOT_OFFSET=0x8b0000
diff --git a/board/habanalabs/dali/btl/primary.S b/board/habanalabs/dali/btl/primary.S
index df27c5a..1d3276a 100644
--- a/board/habanalabs/dali/btl/primary.S
+++ b/board/habanalabs/dali/btl/primary.S
@@ -32,17 +32,13 @@
  * Using scratchpad registers for
  * passing flash layout to CPU SW
  * ------------------------------
- * 25 - BTL_ID
- * 26 - PCIE_PHY_FW_OFFSET
- * 27 - PCIE_PHY_FW_SIZE
+ * 27 - BTL_ID
  * 28 - SPL_OFFSET
  * 29 - SPL_SIZE
  * 30 - UBOOT_OFFSET
  * 31 - UBOOT_SIZE
  */
-cfg 0xCC4B164, BTLID
-cfg 0xCC4B168, PCIE_PHY_FW_OFFSET
-cfg 0xCC4B16C, PCIE_PHY_FW_SIZE
+cfg 0xCC4B16C, BTLID
 cfg 0xCC4B170, SPL_OFFSET
 cfg 0xCC4B174, SPL_SIZE
 cfg 0xCC4B178, UBOOT_OFFSET
@@ -53,7 +49,7 @@
 cfg 0xC441114, PREBOOT_ADDR_H
 
 /* Wake up CPU0, after we put all CPUs in reset */
-cfg 0xC441124, 0
+cfg 0xC441124, 0x0
 cfg 0xC441124, 0x10111
 
 end
